tests/test_writer.py

```diff
@@ -1202,25 +1202,42 @@ def test_attachments():
     b.seek(0)
     reader = PdfReader(b)
     b = None
+    assert reader.attachments == {}
     assert reader._list_attachments() == []
     assert reader._get_attachments() == {}
-    writer.add_attachment("foobar.txt", b"foobarcontent")
-    writer.add_attachment("foobar2.txt", b"foobarcontent2")
-    writer.add_attachment("foobar2.txt", b"2nd_foobarcontent")
+    to_add = [
+        ("foobar.txt", b"foobarcontent"),
+        ("foobar2.txt", b"foobarcontent2"),
+        ("foobar2.txt", b"2nd_foobarcontent"),
+    ]
+    for name, content in to_add:
+        writer.add_attachment(name, content)
 
     b = BytesIO()
     writer.write(b)
     b.seek(0)
     reader = PdfReader(b)
     b = None
-    assert reader._list_attachments() == ["foobar.txt", "foobar2.txt", "foobar2.txt"]
+    assert sorted(reader.attachments.keys()) == sorted({name for name, _ in to_add})
+    assert reader._list_attachments() == [name for name, _ in to_add]
+
+    # We've added the same key twice - hence only 2 and not 3:
     att = reader._get_attachments()
-    assert len(att) == 2
+    assert len(att) == 2  # we have 2 keys, but 3 attachments!
+
+    # The content for foobar.txt is clear and just a single value:
     assert att["foobar.txt"] == b"foobarcontent"
+
+    # The content for foobar2.txt is a list!
     att = reader._get_attachments("foobar2.txt")
     assert len(att) == 1
     assert att["foobar2.txt"] == [b"foobarcontent2", b"2nd_foobarcontent"]
 
+    # Let's do both cases with the public interface:
+    assert reader.attachments["foobar.txt"][0] == b"foobarcontent"
```

Comment:
This interface is not in accordance with the documentation you've provided
Also it should be great to indicate in the documentation that multiple inputs have the same entry name, and justify the systematic [0]

Reply:
> This interface is not in accordance with the documentation you've provided

What do you mean?

> Also it should be great to indicate in the documentation that multiple inputs have the same entry name, and justify the systematic

I wrote: "Attachments have a name, but it might not be unique." I've expanded that a little bit.